gatling-core/src/test/scala/io/gatling/core/session/SessionSpec.scala

```diff
@@ -28,10 +28,17 @@ import io.netty.channel.{ Channel, ChannelFuture, ChannelPromise, EventLoop, Eve
 import io.netty.util.concurrent.{ EventExecutor, Future => NFuture, ProgressivePromise, Promise, ScheduledFuture }
 
 object SessionSpec {
+  private val timer = new Timer
```

Comment:
We shouldn't have state in an object, even if it's for tests.
If FakeEventLoop becomes stateful and hold a resource such as a Timer, it can no longer be an object. It must be a class and we must create an instance in the tests.
The timer should be configured to a daemon Thread, not a user one.
The timer should be cancelled in a `afterAll` method.

Reply:
The state inside the timer feels like the philosophy point of "a tree falls in a forest".

I forgot making the timer thread daemon. Didn't think much because the tests passed. 😅

Reply:
> The state inside the timer feels like the philosophy point of "a tree falls in a forest".

A Timer is a Thread, hence an important resource. I'd rather have it hidden inside a `FakeEventLoop` class and have tests own `FakeEventLoop` instances that they must cancel afterward.

The reason is that I think it would be nice to have FakeEventLoop used in other tests.